File: packages/server/src/infrastructure/__fixtures__/sample.go
package sample

type Point struct {
	X float64
	Y float64
}

func Distance(a Point, b Point) float64 {
	dx := a.X - b.X
	dy := a.Y - b.Y
	return dx*dx + dy*dy
}

type Vec struct {
	X float64
	Y float64
}

func (v Vec) Length() float64 {
	return v.X*v.X + v.Y*v.Y
}

File: packages/server/src/infrastructure/__fixtures__/sample.js
const ORIGIN = { x: 0, y: 0 };

function distance(a, b) {
  return Math.hypot(a.x - b.x, a.y - b.y);
}

class Vec {
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }
  length() {
    return Math.hypot(this.x, this.y);
  }
}

module.exports = { ORIGIN, distance, Vec };
